### PR TITLE
[XLA] Support QuantizeAndDequantizeV4

### PR DESCRIPTION
**Pull request by lgeiger**
@pkanwar23 added  `QuantizeAndDequantizeV4` in 52df91c5634e6c666843849a1c6ff29b3d2676be which is used in `tf.quantization.quantize_and_dequantize_v2` and deprecated `tf.quantization.quantize_and_dequantize` which relied on `QuantizeAndDequantizeV2`.

This PR adds  `QuantizeAndDequantizeV4` support for XLA which can be a simple alias to the `QuantizeAndDequantizeV2` kernel implementation since `QuantizeAndDequantizeV4` only changed the gradient and kept the same forward pass:
https://github.com/tensorflow/tensorflow/blob/e43be76009614be88454d2fdf2fe702acc5bab77/tensorflow/core/kernels/quantize_and_dequantize_op.cc#L419-L422

To make this work with the MLIR bridge, the op also needed to be added to `tf_generated_ops.td`. This file is autogenerated, but unfortunately I couldn't find any documentation on how this is done so I manually added the op for now.

Once this is merged, I am also happy to add support for this in the MLIR TFLite converter which should be straight forward as well.

### CODE REVIEW

**Comment by lgeiger** [comment]
@joker-eph @gbaned Any updates on this PR? It would be cool if this could get reviewed before the 2.5 branch get's cut.

**Comment by lgeiger** [comment]
Any updates on this? Would be great to still get this into the 2.5 branch cut.

**Comment by lgeiger** [comment]
@gbaned any updates on this PR?

**Comment by gbaned** [comment]
> @gbaned any updates on this PR?

@lgeiger This PR is waiting for the reviewer approval. Thanks!

**Review by smit-hinsu — APPROVED** [review]
Thanks for the contributions!

**Comment by lgeiger** [comment]
> Thanks for the contributions!

@smit-hinsu Thanks a lot for the review!

Do you mind also reviewing followup PR that I just pushed: #48410

**Pull request merged**
